va/validation-authority_test.go

```diff
@@ -105,9 +110,21 @@ func simpleSrv(t *testing.T, token string, stopChan, waitChan chan bool, enableT
 		} else if strings.HasSuffix(r.URL.Path, "wait-long") {
 			t.Logf("SIMPLESRV: Got a wait-long req\n")
 			time.Sleep(time.Second * 10)
+		} else if strings.HasSuffix(r.URL.Path, "re-lookup") {
+			t.Logf("SIMPLESRV: Got a redirect req to a valid hostname\n")
+			if currentToken == defaultToken {
+				currentToken = "re-lookup"
+			}
+			http.Redirect(w, r, "http://localhost2/path", 302)
```

Comment:
I would rather if you just make the existing 301 and 302 tests change to a different host, since that's the only difference here.

Also, while I assume `localhost2` is something that the mock DNS resolver is configured to respond for, it would be better to use host names in test that are clearly meant for testing, e.g., hostnames under ".invalid".